master/internal/audit.go

```diff
@@ -83,3 +85,20 @@ func auditLogMiddleware() echo.MiddlewareFunc {
 		}
 	})
 }
+
+func authzAuditLogMiddleware() echo.MiddlewareFunc {
+	return echo.MiddlewareFunc(func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(c echo.Context) (err error) {
+			if !strings.HasPrefix(c.Request().RequestURI, "/api/v1") {
+				fields := log.Fields{"endpoint": c.Request().RequestURI}
+				newCtx := context.WithValue(c.Request().Context(), audit.LogKey{}, fields)
+				c.SetRequest(c.Request().WithContext(newCtx))
+			}
+
+			if err = next(c); err != nil {
+				c.Error(err)
+			}
+			return
```

Comment:
i'm not sure, with echo, what the deal is between returning errs all the way up the stack (just calling `return next(c)`) and catching them and setting them on the context. what is more "echodiomatic"? i think i looked when i wrote the other audit log middleware, but then i notice user auth took a different approach.